scripts/lib/jenkins_generic_job.py

```diff
@@ -1,22 +1,25 @@
 import CIME.wait_for_tests
 from CIME.utils import expect
+from CIME.case import Case
 
 import os, shutil, glob, signal, logging
 
 ###############################################################################
-def cleanup_queue(set_of_jobs_we_created):
+def cleanup_queue(test_root, test_id):
 ###############################################################################
     """
     Delete all jobs left in the queue
     """
-    current_jobs = set(CIME.utils.get_my_queued_jobs())
-    jobs_to_delete = set_of_jobs_we_created & current_jobs
+    for teststatus_file in glob.glob("{}/*{}*/TestStatus".format(test_root, test_id)):
```

Comment:
I think that this should be glob.iglob ?

Reply:
Why iglob?

Reply:
```
glob.iglob(pathname)
Return an iterator which yields the same values as glob() without actually storing them all simultaneously.
```
seems that is all that you need here.

Reply:
Thanks, yes that would be better.

Reply:
Done.